### PR TITLE
Drop elasticsearch and replace search

### PR DESCRIPTION
**Pull request by francisco-leal**
## Summary

Removes the need to use elasticsearch in favour of a native solution for postgres.

### CODE REVIEW

File: app/controllers/api/v1/talent_controller.rb [hunk 1]
```diff
@@ -3,40 +3,35 @@ class API::V1::TalentController < ApplicationController
...
-  end
-
-  # public /
-  def public_index
     talents =
-      if token_id_params.present?
-        Talent.includes(:talent_token, :user).where(talent_tokens: {contract_id: token_id_params})
+      if filter_params[:keyword].present?
+        query = <<~SQL
+          SELECT "users".* FROM "users"
```
Comment (RubenSousaDinis) [inline app/controllers/api/v1/talent_controller.rb:9]: I think you want a list of talents and not users here.

File: app/controllers/api/v1/talent_controller.rb [hunk 2]
```diff
@@ -3,40 +3,29 @@ class API::V1::TalentController < ApplicationController
...
-    }, status: :ok
-  end
-
-  # public /
-  def public_index
     talents =
-      if token_id_params.present?
-        Talent.includes(:talent_token, :user).where(talent_tokens: {contract_id: token_id_params})
+      if filter_params[:keyword].present?
+        filter_query = ActiveRecord::Base.sanitize_sql_array(["SIMILARITY(COALESCE(display_name, username), :keyword) >= 0.3", keyword: "fran"])
```
Comment (RubenSousaDinis) [inline app/controllers/api/v1/talent_controller.rb:8]: hardcoded keyword

Comment (RubenSousaDinis) [inline app/controllers/api/v1/talent_controller.rb:8]: I think it would be cool to have this value configurable to test results easier

Comment (RubenSousaDinis) [reply app/controllers/api/v1/talent_controller.rb]: yes! But already understood 0.3 is the recommended setting. All good to keep it as is

**Review by RubenSousaDinis — APPROVED** [review]

**Review by verdefred — APPROVED** [review]

**Pull request merged**